src/main/java/com/palantir/refreshable/DefaultRefreshable.java

```diff
@@ -159,14 +148,57 @@ private void preSubscribeLogging() {
         }
     }
 
+    @Override
+    public <R> Refreshable<R> map(Function<? super T, R> function) {
+        readLock.lock();
+        try {
+            R initialChildValue = function.apply(current);
+            SettableRefreshable<R> child = createSettableChild(initialChildValue);
+
+            // Contract for Cleaner#register states that the Runnable should never reference the object being registered
+            Disposable disposable = registerMapSubscriber(function, new WeakReference<>(child));
+
+            REFRESHABLE_CLEANER.register(child, disposable::dispose);
+
+            return child;
+        } finally {
+            readLock.unlock();
+        }
+    }
+
+    private <R> SettableRefreshable<R> createSettableChild(R initialChildValue) {
```

Comment:
This is interesting, and raises something I've been thinking about:

I don't think we necessarily care that the children are settable refreshables, in fact we should probably wrap the returned refreshable with a thin shim to remove implementation type information (we would never want someone to cast to settablerefreshable and update a mapped refreshable!). For a separate PR/feature though!
We happen to use the same codepath out of convenience, so I think the previous `DefaultRefreshable` type (for private implementation details) was reasonable.